src/main/java/network/brightspots/rcv/GuiConfigController.java

```diff
@@ -518,71 +521,95 @@ public void buttonClearDatePickerContestDateClicked() {
   public void buttonAddCvrFileClicked() {
-    CvrSource cvrSource =
-        new CvrSource(
-            getTextOrEmptyString(textFieldCvrFilePath),
-            getTextOrEmptyString(textFieldCvrFirstVoteCol),
-            getTextOrEmptyString(textFieldCvrFirstVoteRow),
-            getTextOrEmptyString(textFieldCvrIdCol),
-            getTextOrEmptyString(textFieldCvrPrecinctCol),
-            getTextOrEmptyString(textFieldCvrOvervoteDelimiter),
-            getProviderChoice(choiceCvrProvider).getInternalLabel(),
-            getTextOrEmptyString(textFieldCvrContestId),
-            getTextOrEmptyString(textFieldCvrOvervoteLabel),
-            getTextOrEmptyString(textFieldCvrUndervoteLabel),
-            getTextOrEmptyString(textFieldCvrUndeclaredWriteInLabel),
-            checkBoxCvrTreatBlankAsUndeclaredWriteIn.isSelected()
-        );
-    if (ContestConfig.passesBasicCvrSourceValidation(cvrSource)) {
-      tableViewCvrFiles.getItems().add(cvrSource);
-      textFieldCvrFilePath.clear();
-    }
+    List<String> cvrPaths = Arrays.stream(getTextOrEmptyString(textFieldCvrFilePath).split(
+        CVR_FILE_PATH_DELIMITER)).toList();
+    List<String> failedFilePaths = new ArrayList<>();
+    String firstVoteColumnIndex = getTextOrEmptyString(textFieldCvrFirstVoteCol);
+    String firstVoteRowIndex = getTextOrEmptyString(textFieldCvrFirstVoteRow);
+    String idColumnIndex = getTextOrEmptyString(textFieldCvrIdCol);
+    String precinctColumnIndex = getTextOrEmptyString(textFieldCvrPrecinctCol);
+    String overvoteDelimiter = getTextOrEmptyString(textFieldCvrOvervoteDelimiter);
+    String provider = getProviderChoice(choiceCvrProvider).getInternalLabel();
+    String contestId = getTextOrEmptyString(textFieldCvrContestId);
+    String overvoteLabel = getTextOrEmptyString(textFieldCvrOvervoteLabel);
+    String undervoteLabel = getTextOrEmptyString(textFieldCvrUndervoteLabel);
+    String undeclaredWriteInLabel = getTextOrEmptyString(textFieldCvrUndeclaredWriteInLabel);
+    boolean treatBlankAsUndeclaredWriteIn = checkBoxCvrTreatBlankAsUndeclaredWriteIn.isSelected();
+
+    cvrPaths.forEach(filePath -> {
+      CvrSource cvrSource =
+          new CvrSource(
+              filePath,
+              firstVoteColumnIndex,
+              firstVoteRowIndex,
+              idColumnIndex,
+              precinctColumnIndex,
+              overvoteDelimiter,
+              provider,
+              contestId,
+              overvoteLabel,
+              undervoteLabel,
+              undeclaredWriteInLabel,
+              treatBlankAsUndeclaredWriteIn
+          );
+      if (ContestConfig.passesBasicCvrSourceValidation(cvrSource)) {
+        tableViewCvrFiles.getItems().add(cvrSource);
+      } else {
+        failedFilePaths.add(filePath);
+      }
+    });
+    // If any entries failed validation, preserve them in the text box so the user can try again
+    textFieldCvrFilePath.setText(String.join(CVR_FILE_PATH_DELIMITER, failedFilePaths));
```

Comment:
Is this standard behavior elsewhere, or just something you thought of on your own? I can see how it might be useful, but it may be confusing if we don't tell the user explicitly what just happened (x files added successfully; y invalid files remain).

Reply:
Yes, it's standard to leave the values they previously entered in place if validation failed (I believe this was specifically request a while back).

In practice, we should never have a situation where something is invalid for only one of the items in the list (since they're all going to have the same properties; the one exception is if you do something like `;;;;;;` between other valid entries), but even if that were the case, the logging console should make it clear below what happened.

Essentially what this ends up doing is preserving everything in the Path input if validation fails for one or more of the properties (like a missing first vote column value), which I think is a good and intuitive thing.

Reply:
Yeah, I just think it might be more intuitive to make the code explicitly all or none. If somehow we successfully added some but not all, that would be kind of weird.